shared/wallets/transaction-details/index.js

```diff
@@ -45,64 +45,71 @@ export type Props =
   | NotLoadingProps
   | {|loading: true, onBack: () => void, onLoadPaymentDetail: () => void, title: string|}
 
-type CounterpartyIconProps = {|
-  onShowProfile: string => void,
-  counterparty: string,
-  counterpartyType: Types.CounterpartyType,
+type PartyAccountProps = {|
+  accountID: ?Types.AccountID,
+  accountName: string,
 |}
 
-export const CounterpartyIcon = (props: CounterpartyIconProps) => {
-  const size = 32
-  switch (props.counterpartyType) {
-    case 'keybaseUser':
-      return (
-        <Kb.Avatar
-          onClick={() => props.onShowProfile(props.counterparty)}
-          username={props.counterparty}
-          size={size}
-        />
-      )
-    case 'stellarPublicKey':
-      return <Kb.Icon type="icon-placeholder-secret-user-32" style={{height: size, width: size}} />
-    case 'otherAccount':
-      return <Kb.Icon type="icon-wallet-32" style={{height: size, width: size}} />
-    default:
-      /*::
-      declare var ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove: (counterpartyType: empty) => any
-      ifFlowErrorsHereItsCauseYouDidntHandleAllActionTypesAbove(props.counterpartyType);
-      */
-      return null
-  }
+const PartyAccount = (props: PartyAccountProps) => {
```

Comment:
did a buncha refactoring here -- didn't change anything as you can see from the storyshots PR coming before all the refactoring